client/client.go

```diff
@@ -16,24 +16,49 @@ type Client struct {
 	endpoint   string
 	httpClient *http.Client
 	logger     *logger.Logger
+	token      string
 }
 
 // NewClient returns a reference to a Client.
-func NewClient(endpoint string, logger *logger.Logger) *Client {
+func NewClient(endpoint string, token string, logger *logger.Logger) *Client {
 	return &Client{
 		httpClient: http.DefaultClient,
 		endpoint:   endpoint,
+		token:      token,
 		logger:     logger,
 	}
 }
 
 // NewAuthorizedRequest returns a new GraphQL request with the
 // authorization headers set for CircleCI auth.
-func NewAuthorizedRequest(token, query string) *Request {
-	req := NewRequest(query)
-	req.Header.Set("Authorization", token)
-	req.Header.Set("User-Agent", version.UserAgent())
-	return req
+func (cl *Client) NewAuthorizedRequest(query string) (*Request, error) {
+	if cl.token == "" {
+		return nil, errors.New(`please set a token with 'circleci setup'
+You can create a new personal API token here:
+https://circleci.com/account/api`)
+	}
+
+	request := &Request{
+		Query:     query,
+		Variables: make(map[string]interface{}),
+		Header:    make(map[string][]string),
+	}
+
+	request.Header.Set("Authorization", cl.token)
+	request.Header.Set("User-Agent", version.UserAgent())
+	return request, nil
+}
+
+// NewUnauthorizedRequest returns a new GraphQL request without any authorization header.
+func (cl *Client) NewUnauthorizedRequest(query string) (*Request, error) {
```

Comment:
Nice! I like these (`NewAuthorizedRequest`, `NewUnauthorizedRequest`) constructors.  It seems like they don't make use of `*Client` much. It might make sense to make these package-level functions instead of methods on `Client`, and pass `token` as an arg to `NewAuthorizedRequest`

In the future you could use the [function options pattern](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) to support setting variables and headers as well.